File: ios/VoiceHostSoftphone/Models/CallState.swift
import Foundation

enum CallState: Equatable {
    case idle
    case connecting
    case outgoing(number: String)
    case ringing(number: String)
    case incoming(number: String, displayName: String?)
    case earlyMedia(number: String)
    case connected(number: String)
    case held(number: String)
    case ended(reason: String?)

    var isInCall: Bool {
        switch self {
        case .outgoing, .ringing, .incoming, .earlyMedia, .connected, .held:
            return true
        default:
            return false
        }
    }
}

File: flutter/lib/models/call_state.dart
enum CallPhase {
  idle,
  outgoing,
  ringing,
  incoming,
  earlyMedia,
  connected,
  held,
  ended,
}

class PhoneCallState {
  const PhoneCallState({
    required this.phase,
    this.number,
    this.displayName,
    this.reason,
  });

  const PhoneCallState.idle() : this(phase: CallPhase.idle);

  final CallPhase phase;
  final String? number;
  final String? displayName;
  final String? reason;

  bool get isInCall => switch (phase) {
        CallPhase.outgoing ||
        CallPhase.ringing ||
        CallPhase.incoming ||
        CallPhase.earlyMedia ||
        CallPhase.connected ||
        CallPhase.held => true,
        _ => false,
      };

  bool get isConnected =>
      phase == CallPhase.connected || phase == CallPhase.held;
}
